ekncontent/ekncontent/eknc-content-object-model.c

```diff
@@ -252,6 +252,7 @@ eknc_content_object_model_constructed (GObject *object)
                                          (guchar *) &MOCKED_OBJECTS,
                                          sizeof (MOCKED_OBJECTS));
       MOCKED_OBJECTS++;
+      g_clear_pointer(&priv->ekn_id, g_free);
```

Comment:
This is really not necessary:
 - `g_free()` is `NULL`-safe
 - you're changing the pointer right after that
 - `g_strconcat()` will return `NULL` on error

I've seen this pattern elsewhere, and I'd rather we stopped using `g_clear_pointer()` when it's entirely unnecessary.

Reply:
OK, will change, but why not just always use `g_clear_pointer()` and eliminate the cognitive load?

Reply:
Well, mostly because it's pointless line noise, and `g_clear_pointer()` is not free, as it needs to do an atomic pointer swap, which is a memory barrier and a synchronisation point.

Clearing the pointer is only useful if you're nullifying it, not replacing it with something else immediately afterwards. For GObject we have `g_steal_object()`, but that's because it's easier to use as it does not have an additional argument for the free function. Maybe we could have a `g_steal_pointer()` in GLib.